linode/manager
# M3-2122 Upgrade to Webpack 4

## Description

Please see [this PR first.](https://github.com/martinmckenna/manager/pull/1/files)

tl:dr webpack is for the birds

Upgrading from Webpack 3 to 4 due to Github security update informing us of a security issue with our current version of webpack-dev-server.

So basically, I updated webpack-dev-server, learned that the new version requires Webpack 4 and then had to update a bunch of other things to be compatible with Webpack 4.

## Note to Reviewers

#### Changed
* Upgraded to Webpack 4
  * see [here](https://webpack.js.org/migrate/4) for migration notes
  * you'll notice that many plugins are now enabled by default in Webpack and others had to be either upgraded, or replaced with other versions
* Changed import statements
  * [see here for issue](https://github.com/TypeStrong/ts-loader/issues/751). Unfortunately, we cannot re-export modules and will instead have to import/export them again
      * that being said, we should move away from this pattern entirely, and instead create type files in which we export interfaces and types

## To Test
* Run every command to make sure nothing is broken
  * `yarn build`, `yarn start`, `yarn test`, `yarn storybook`
  * make sure to `rm -rf node_modules` first
